google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryResponse.java

```diff
@@ -33,6 +34,8 @@
 
   abstract long getTotalRows();
 
+  abstract ImmutableList<BigQueryError> getErrors();
```

Comment:
I'm not sure about plumbing errors here, since it doesn't apply to reading results directly from a table. I guess no matter what we want a way to expose these from the `query()` method so we'll want something like `QueryResults`, anyway. I guess it makes sense to add here.

Reply:
What do you think `QueryResults` should look like? How would it relate to the existing `QueryResult` (without the 's')?

You're right that this error doesn't apply for reading directly from table, but we're not using this class to read right? `getQueryResults` returns this when the query completes; `QueryResponse` is a bit of misnomer.

Reply:
My ideal is to have a shared class for listing table data so that we can expose properties like `totalRows()` on both, but honestly the split between the two (query results and table listing) does make some sense. I was just thinking aloud.

I do like the current name `QueryResults` better than `QueryResponse`.

Reply:
I see what you're saying. I think having a shared class make sense; it'll let users inject schema when iterating table too.

I'm thinking of something like this.
- Keep `QueryResponse` an "internal" class as it is. Just use it to wait for query and get errors.
- Rename `QueryResult` -> `TableResult` or something like it. Use it for both `query` and `listTableData`.
  - Currently we read `totalRows` from `getQueryResults` RPC, we'll change it to read from `tabledata.list` instead.

If this sounds about right to you, let's get this PR merged for now. I'll work on these in a follow up.

Reply:
Yeah, that sounds about right.